hack/app-sre/kustomization.yaml

```diff
@@ -0,0 +1,46 @@
+# This kustomize configuration deploys what is required for the hive-operator
+# *without* OLM.
+# Except:
+# It does not contain a HiveConfig.
+
+# Adds namespace to all resources.
+namespace: hive
+
+# The CRD resources are synced automatically by the `build-app-sre-template` make target.
+# The other resources should be curated manually.
+resources:
+- ../../config/namespace.yaml
+- ../../config/operator/operator_role.yaml
+- ../../config/operator/operator_role_binding.yaml
+- ../../config/operator/operator_deployment.yaml
+- ../../config/crds/hiveinternal.openshift.io_clustersyncleases.yaml
+- ../../config/crds/hiveinternal.openshift.io_clustersyncs.yaml
+- ../../config/crds/hiveinternal.openshift.io_fakeclusterinstalls.yaml
+- ../../config/crds/hive.openshift.io_checkpoints.yaml
+- ../../config/crds/hive.openshift.io_clusterclaims.yaml
+- ../../config/crds/hive.openshift.io_clusterdeployments.yaml
+- ../../config/crds/hive.openshift.io_clusterdeprovisions.yaml
+- ../../config/crds/hive.openshift.io_clusterimagesets.yaml
+- ../../config/crds/hive.openshift.io_clusterpools.yaml
+- ../../config/crds/hive.openshift.io_clusterprovisions.yaml
+- ../../config/crds/hive.openshift.io_clusterrelocates.yaml
+- ../../config/crds/hive.openshift.io_clusterstates.yaml
+- ../../config/crds/hive.openshift.io_dnszones.yaml
+- ../../config/crds/hive.openshift.io_hiveconfigs.yaml
+- ../../config/crds/hive.openshift.io_machinepoolnameleases.yaml
+- ../../config/crds/hive.openshift.io_machinepools.yaml
+- ../../config/crds/hive.openshift.io_selectorsyncidentityproviders.yaml
+- ../../config/crds/hive.openshift.io_selectorsyncsets.yaml
+- ../../config/crds/hive.openshift.io_syncidentityproviders.yaml
+- ../../config/crds/hive.openshift.io_syncsets.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+# Use app-sre-supplied variables to pull, by digest, the image for the current commit
+images:
+- digest: ${IMAGE_DIGEST}
+  name: registry.ci.openshift.org/openshift/hive-v4.0:hive
+  newName: ${REGISTRY_IMG}
+# Label all the deployed artifacts with the commit hash so we can tell everything is in sync
+commonLabels:
+  managed.openshift.io/gitHash: ${IMAGE_TAG}
```

Comment:
this will cause all resources to be applied on every deployment. i don't think this is desired. smells like fear of OLM ;)

Reply:
> this will cause all resources to be applied on every deployment.

Because appsre is comparing old/new and skipping resources if they match, or because `oc apply` is doing that?

> i don't think this is desired.

Why not? IIUC `oc apply` does it with an update, not a delete/create, so it should be non-invasive.

> smells like fear of OLM ;)

I mean, yes, I'm doing it deliberately as an answer to the OLM feature where, if the CSV is applied successfully, you can be sure* that all the resources are at the right version.

We're going to need an SOP for debugging problems in this new world, and "make sure all the resources are at the right version" needs to be a doable step. Do you have suggestions for other ways we could do that?

*for most values of "sure"

Reply:
app-interface takes care of this for you. if a resource can not be applied, the deployment will fail. we don't want all resources applied every time. there is not another service we run that does this.

Reply:
> we don't want all resources applied every time

Can you explain why? I most certainly thought we did want that. What is the downside or risk of just always applying everything? Why wouldn't we just let `kubectl apply -f` figure it out?

Reply:
I would assume that is what app-interface is doing, kubectl apply everything, if it fails fail the deployment. Is this correct @maorfr ?

Reply:
I think I'd agree with @maorfr, we shouldn't need to verify the deployed version, our gitops tooling should do that for us and there shouldn't be any need to update resources that haven't changed.

Reply:
Okay, I've taken it out. Easy enough to put back later if we find we need it.